docs/UserGuide.md

```diff
@@ -242,68 +230,110 @@ Expected behaviour upon success:
 --------------------------------------------------------------------------------------------------------------------
-### Viewing all vendors: `vendor list`
-View all vendors in a list format.
+
+### Filtering guests: `guest filter`
+
+Filters the guest list with keywords.
 
 ```text
-vendor list
+guest filter [n/NAME] [p/PHONE_NUMBER] [e/EMAIL] [a/ADDRESS] [r/RSVP_STATUS] [d/DIETARY REQUIREMENTS] [tn/TABLE_NUMBER] [t/TAG...]
 ```
 
+The following parameters are optional: `p/PHONE_NUMBER e/EMAIL a/ADDRESS r/RSVP_STATUS d/DIETARY_REQUIREMENTS tn/TABLE_NUMBER t/TAG...`
+
+>Tips:
+><br>
+>- Parameters can be in any order.
+   ><br>
+>- A filter command can have any number of tags (including 0).
+   ><br>
+>- Specifying an empty parameter will filter out guests with unfilled specified parameter.
+   ><br>
+>- Mandatory parameters (e.g. NAME, RSVP_STATUS, etc.) cannot be filtered with empty inputs.
+   ><br>
+>- Guest filter command requires at least 1 parameter
+   ><br>
+>- Refer to [Appendix A](#appendix-a-acceptable-values-for-parameters) for more details on the acceptable values for the parameters.
+
+Examples:
+- `guest filter n/Bob p/91234567 a/Blk 123 r/no`: guests that are filtered need to have Name, Phone number, Address & Rsvp status that matches any of the stated keywords
+- `guest filter n/Keith p/92354567 r/yes`
+- `guest filter n/Jane Tan tn/9`:
+- `guest filter n/John Doe p/98765432 e/john@doe.com a/Street 456 r/unknown d/vegetarian tn/13`
+
 Expected behaviour upon success:
-- Displays a list of all vendor names and their respective indexes. (Example: 1. John FLORAL, 2. Sally Anne PHOTOGRAPHER)
+- Displays a list of guests that has parameters that match any of the keywords specified for that parameter.
 
 Expected behaviour upon failure:
-<br>(refer to Appendix A: Expected behaviour upon general failure)
+- Empty name: `guest filter n/ ` Displays error message "Cannot filter for empty compulsory field"
+- Empty rsvp status: `guest filter r/ ` Displays error message "Cannot filter for empty compulsory field"
+- No parameter: `guest filter` Displays error message "No prefix was found in the command!" followed by instruction on guest filter usage
 
+<span style="font-size: 0.4em;">[Back to Top](#wedlog-user-guide)</span>
 --------------------------------------------------------------------------------------------------------------------
-### Viewing a specific vendor: `vendor view`
-View a specific vendor using a specified index.
+
+### Filtering vendors: `vendor filter`
+
+Filters the vendor list with keywords.
 
 ```text
-vendor view INDEX
+vendor filter [n/NAME] [p/PHONE_NUMBER] [e/EMAIL] [a/ADDRESS] [t/TAG...]
 ```
 
-Acceptable values for INDEX
-- A positive integer
+The following parameters are optional: `n/NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS t/TAG...`
+
+>Tips:
+><br>
+>- Parameters can be in any order.
+   ><br>
+>- A filter command can have any number of tags (including 0).
```

Comment:
i know we're still ironing out the implementation of the filter, but iirc, the command allows for any number of any type of param, not just tags?

Reply:
oh ok i see the difference now. thanks for the clear explanation!